azurelinuxagent/common/protocol/wire.py

```diff
@@ -778,18 +776,30 @@ def update_host_plugin(self, container_id, role_config_name):
             self._host_plugin.update_container_id(container_id)
             self._host_plugin.update_role_config_name(role_config_name)
 
-    def update_goal_state(self, force_update=False, silent=False):
+    def update_goal_state(self, silent=False):
         """
-        Updates the goal state if the incarnation or etag changed or if 'force_update' is True
+        Updates the goal state if the incarnation or etag changed
         """
         try:
-            if force_update and not silent:
+            if not silent:
                 logger.info("Forcing an update of the goal state.")
 
-            if self._goal_state is None or force_update:
-                self._goal_state = GoalState(self, silent=silent)
-            else:
-                self._goal_state.update(silent=silent)
+            self._goal_state.update(silent=silent)
```

Comment:
I think we need to keep a call to reset here (minus the check for the force flag). This is needed by the initialization of the goal state on line 430 that you pointed out

```
            if self._goal_state is None:
                self._goal_state = GoalState(self, silent=silent)
            else:
                self._goal_state.update(silent=silent)

```

Reply:
I added the None check back